File: pkg/server/pokete/msg/fight/fight_decision.go
package fight

import "github.com/lxgr-linux/pokete/bs_rpc/msg"

const FightDecisionType msg.Type = "pokete.fight.fightDecision"

type Result uint

const (
	Result_Attack Result = iota
	Result_RunAway
	Result_Item
	Result_ChoosePoke
)

type FightDecision struct {
	msg.BaseMsg
	Result Result `json:"result"`
	Attack string `json:"attack"`
	Item   string `json:"item"`
}

func (r FightDecision) GetType() msg.Type {
	return FightDecisionType
}

File: pokete_classes/multiplayer/msg/fight/fight_decision.py
from typing import TypedDict

import bs_rpc

FIGHT_DECISION_TYPE = "pokete.fight.fightDecision"


class FightDecisionData(TypedDict):
    result: int
    attack: str | None
    item: str | None


class FightDecision(bs_rpc.Body):
    def __init__(self, data: FightDecisionData):
        super().__init__(FIGHT_DECISION_TYPE, data)
